### PR TITLE
Improving dependency path display for warnings and errors

### LINKED ISSUE #5824: Move error codes with warning numbers to error range
NU1607 and NU1606 are errors but they are in the warning code range. They should be moved to the error range.

### LINKED ISSUE #5852: Dependency paths display incorrect version ranges
Dependency paths for downgrade and version conflict messages with PackageReference projects display the incorrect version range for a dependency.

#### Expected:
The version range used in the nuspec

#### Actual:

The range *>= {resolved version}* is used

A new format that includes both the resolved version and the range might make these messages more useful. Alternatively the resolved version could be used for all packages except the conflicts which could show the range.

Related: https://github.com/NuGet/Home/issues/5463

### LINKED ISSUE #5861: Downgrade warning message lists the min range version instead of the version actually used
The downgrade message reads as the version downgraded to and the lower bound of the range it violates.

`Detected package downgrade: NuGet.Versioning from 4.0.0 to 3.5.0.`

In the above example 3.5.0 is the min version on the package that was resolved. This may not be the actual version used.

If the project references NuGet.Versioning with `3.5.*` then `3.5.0` will be displayed, however `3.5.9` might be the actual version.

The message should display the actual version, and then provide the ranges as part of the dependency paths listed below it.

### PR DESCRIPTION
**Pull request by emgarten**
Improving dependency path display for warnings and errors

This fixes how dependency paths are displayed. Currently they are written out as:
`a (>= 1.0.0) -> b (>= 2.0.0)`

However the ranges here are made up based on the resolved version and are not the actual nuspec dependency ranges. For locked ranges such as `(= 2.0.0)` the range `(>= 2.0.0)` is shown if 2.0.0 is found.

To improve this the path will now be displayed using the actual version, and in some cases the actual range for the last node displayed since this is where the conflict is.

GetPath:
`a -> b 2.0.0 -> c 1.0.0`

GetPathWithLastRange:
`a -> b 2.0.0 -> c (>= 1.0.0)`

Projects will no longer display a version or range, this information is not relevant since version ranges are never applied to projects and there can only be 1 instance of a project in the graph.

Fixes https://github.com/NuGet/Home/issues/5861
Fixes https://github.com/NuGet/Home/issues/5852
Fixes https://github.com/NuGet/Home/issues/5824

### CODE REVIEW

**Review by jainaashish — APPROVED** [review]
Good work! :shipit:

File: src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs [hunk 1]
```diff
@@ -159,16 +168,6 @@ public enum NuGetLogCode
...
-        /// Circular dependency.
-        /// </summary>
-        NU1606 = 1606,
-
-        /// <summary>
-        /// Version conflict.
-        /// </summary>
-        NU1607 = 1607,
-
-        /// <summary>
```
Comment (mishra14) [inline src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs]: Should we leave  comment here asking people to not create these?

**Review by mishra14 — APPROVED** [review]

File: src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs]: I'm with @mishra14 here, we should not delete these but rather mark them as deprecated.

Comment (emgarten) [reply src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs]: I'll add comments

**Pull request merged**